nmosauth/auth_server/models.py

```diff
@@ -69,27 +69,36 @@ class OAuth2Token(db.Model, OAuth2TokenMixin):
 
     id = db.Column(db.Integer, primary_key=True)
     user_id = db.Column(
-        db.Integer, db.ForeignKey('user.id', ondelete='CASCADE'))
-    user = db.relationship('User')
+        db.Integer, db.ForeignKey('admin_user.id', ondelete='CASCADE'))
+    admin_user = db.relationship('AdminUser')
     access_token = db.Column(db.String(255), nullable=False, unique=False)
 
     def is_refresh_token_expired(self):
         expires_at = self.issued_at + self.expires_in * 2
         return expires_at < time.time()
 
 
-class AccessRights(db.Model):
-    __tablename__ = 'access_rights'
+class ResourceOwner(db.Model):
+    __tablename__ = 'resource_owner'
 
     id = db.Column(db.Integer, primary_key=True)
     user_id = db.Column(
-        db.Integer, db.ForeignKey('user.id', ondelete='CASCADE'))
-    user = db.relationship('User')
+        db.Integer, db.ForeignKey('admin_user.id', ondelete='CASCADE'))
+    admin_user = db.relationship('AdminUser')
+
+    username = db.Column(db.String(40), unique=True, nullable=False)
+    password = db.Column(db.String(20))
     is04 = db.Column(db.String(25))
     is05 = db.Column(db.String(25))
 
+    def get_user_id(self):
+        return self.id
+
+    def check_password(self, password):
+        return password == self.password
```

Comment:
Are the passwords hashed at all in the database?

Reply:
They are in this commit: c91d10835befa3273cd5c0da365ec72008ea62f9

Reply:
Is that a work in progress?

Reply:
I've yet to fix the tests to make it work bu the functionality all seems to work fine yeah

Reply:
Ok. Would be good to see that included as soon as possible.